ndlarflow/rootToRootConversion.C

```diff
@@ -0,0 +1,455 @@
+    outgoingTree->Branch("run", &run);
+    outgoingTree->Branch("subrun", &subrun);
+    outgoingTree->Branch("event", &event);
+    outgoingTree->Branch("event_start_t", &event_start_t);
+    outgoingTree->Branch("event_end_t", &event_end_t);
+    outgoingTree->Branch("unix_ts", &unix_ts);
+    outgoingTree->Branch("x", &x);
+    outgoingTree->Branch("y", &y);
+    outgoingTree->Branch("z", &z);
+    outgoingTree->Branch("ts", &ts);
+    outgoingTree->Branch("E", &E);
+    outgoingTree->Branch("charge", &charge);
+    if ( isMC ) {
+        outgoingTree->Branch("hit_pdg", &hit_pdg);
+        outgoingTree->Branch("hit_segmentID", &hit_segmentID);
+        outgoingTree->Branch("hit_particleID", &hit_particleID);
+        outgoingTree->Branch("hit_particleIDLocal", &hit_particleIDLocal);
+        outgoingTree->Branch("hit_vertexID", &hit_vertexID);
+        outgoingTree->Branch("hit_packetFrac", &hit_packetFrac);
+        outgoingTree->Branch("mcp_px", &mcp_px);
+        outgoingTree->Branch("mcp_py", &mcp_py);
+        outgoingTree->Branch("mcp_pz", &mcp_pz);
+        outgoingTree->Branch("mcp_id", &mcp_id);
+        outgoingTree->Branch("mcp_idLocal", &mcp_idLocal);
+        outgoingTree->Branch("mcp_nuid", &mcp_nuid);
+        outgoingTree->Branch("mcp_vertex_id", &mcp_vertex_id);
+        outgoingTree->Branch("mcp_pdg", &mcp_pdg);
+        outgoingTree->Branch("mcp_mother", &mcp_mother);
+        outgoingTree->Branch("mcp_energy", &mcp_energy);
+        outgoingTree->Branch("mcp_startx", &mcp_startx);
+        outgoingTree->Branch("mcp_starty", &mcp_starty);
+        outgoingTree->Branch("mcp_startz", &mcp_startz);
+        outgoingTree->Branch("mcp_endx", &mcp_endx);
+        outgoingTree->Branch("mcp_endy", &mcp_endy);
+        outgoingTree->Branch("mcp_endz", &mcp_endz);
+        outgoingTree->Branch("nuvtxx", &nuvtxx);
+        outgoingTree->Branch("nuvtxy", &nuvtxy);
+        outgoingTree->Branch("nuvtxz", &nuvtxz);
+        outgoingTree->Branch("nupx", &nupx);
+        outgoingTree->Branch("nupy", &nupy);
+        outgoingTree->Branch("nupz", &nupz);
+        outgoingTree->Branch("nue", &nue);
+        outgoingTree->Branch("nuID", &nuID);
+        outgoingTree->Branch("vertex_id", &vertex_id);
+        outgoingTree->Branch("nuPDG", &nuPDG);
+        outgoingTree->Branch("mode", &mode);
+        outgoingTree->Branch("ccnc", &ccnc);
+    }
+
+    int thisRun=0;
+    int thisSubRun=0;
+    int thisEvent=0;
+
+    unsigned long sum_matches=0;
+    std::vector<short> all_matches;
+    std::vector<int>   all_hit_pdg;
+    std::vector<long>  all_hit_segmentID;
+    std::vector<long>  all_hit_particleID;
+    std::vector<long>  all_hit_particleIDLocal;
+    std::vector<long>  all_hit_vertexID;
+    std::vector<float> all_hit_packetFrac;
+
+    for (unsigned int idx=0; idx<=NEvents; ++idx) {
+        std::cout << in_run << ":" << in_subrun << ":" << in_event << ":" << in_subevent << std::endl;
```

Comment:
To reduce the size of the screen/log printout, could we just print this out for every 10 or 100 events or so, and not every event?

Reply:
Yeah fair, and since this is a loop on subevents, we may be in the case of several per event. So, I went with the 100 option for now.